apps/ui/src/components/search-box.tsx

```diff
@@ -0,0 +1,19 @@
+import { Search } from "../icons/search";
+
+interface SearchBoxProps {
+  handleSearch: (value: string) => void;
+}
+
+export function SearchBox({ handleSearch }: SearchBoxProps) {
+  return (
+    <div className="flex items-center gap-2 border border-accent  py-1 px-2 rounded-md">
+      <Search width={15} height={15} />
+      <input
+        type="text"
+        className="bg-transparent focus:outline-none"
+        placeholder="Search"
+        onChange={({ target: { value } }) => handleSearch(value)}
```

Comment:
Cool way to destructure the event handler.

Reply:
I think we should do `handleSearch(value.trim())`. Otherwise, the search will get matched against any empty string - like ` ` or `               `.

Reply:
I agree, great idea. I would just add `trim()` inside `export function traceNameIncludes(trace: Trace, searchValue: string): boolean {
  return rootSpanName({ trace })
    .toLowerCase()
    .includes(searchValue.toLowerCase().trim());
}
` function as if we add directly to setState() it will limit user from adding spaces on search box